buying_selling/users/api/views.py

```diff
@@ -70,3 +72,43 @@ def perform_update(self, serializer):
 #             serializer.save()
 #             return Response(serializer.data)
 #         return Response(serializer.errors)
+
+
+class SavedPostViewset(ModelViewSet):
+
+    serializer_action_classes = {
+        'create': SavedPostCreateSerializer,
+        'list': SavedPostListSerializer,
+    }
+    queryset = SavedPosts.objects.all()
+
+    permission_classes_by_action = {
+        'create': [IsAuthenticated],
+        'list': [IsAuthenticated, IsOwnerOrReadOnly],
+        'destroy': [IsAuthenticated, IsOwnerOrReadOnly],
+    }
+
+    def get_serializer_class(self):
+        return self.serializer_action_classes[self.action]
+
+    def get_permissions(self):
+        try:
+            return [permission() for permission in self.permission_classes_by_action[self.action]]
+        except KeyError:
+            return [permission() for permission in self.permission_classes]
+
+    def perform_create(self, serializer):
+        payload = jwt_decoder(self.request.headers['Authorization'].split()[1])
+        serializer.save(author_id=payload['user_id'])
+
+    def list(self, request, *args, **kwargs):
+        payload = jwt_decoder(self.request.headers['Authorization'].split()[1])
+        current_user = MyUser.objects.get(id=payload['user_id'])
+        queryset = self.filter_queryset(SavedPosts.objects.filter(author=current_user))
+        page = self.paginate_queryset(queryset)
+        if page not in None:
```

Comment:
Iterating over a NoneType object isn't possible. Is there anything else that you wanted to achieve?